### ISSUE TITLE
Add ability to track inputs/outputs of CNN model

### ISSUE THREAD

**Issue by aamster** [AllenInstitute/DeepCell#2]
Currently, tracking of inputs/outputs of the CNN model is not good making it so that it is not easy/ not possible to reproduce results.

We need to track:
- Git commit
- model hyperparameters (architecture, learning rate, dropout rate, number of frozen layers, l2 regularization, data preprocessing (what augmentations were used, cropping size, etc), loss function, early stopping, number of epochs, learning rate decay, ...)
- Data that went into the model and how to refetch that data
- model outputs: CV performance (train & val loss, accuracy), and plots of epoch vs loss, epoch vs accuracy for train, val
- model weights

We should also make it so that it is easy to fetch the inputs for a model and reproduce results.

**Comment by aamster** [comment]
Included as part of #18
